TargetComposition/SymBuilder/ImportSymbols.h

```diff
@@ -0,0 +1,557 @@
+    //
+    HRESULT ImportPointer(_In_ ULONG symIndex, _Out_ ULONG64 *pBuilderId, _In_ ULONG64 parentId = 0);
+
+    // ImportTypedef():
+    //
+    // Imports the given typedef into the symbol builder.
+    //
+    HRESULT ImportTypedef(_In_ ULONG symIndex, _Out_ ULONG64 *pBuilderId, _In_ ULONG64 parentId = 0);
+
+    // ImportFunctionType():
+    //
+    // Imports the given function type into the symbol builder.
+    //
+    HRESULT ImportFunctionType(_In_ ULONG symIndex, _Out_ ULONG64 *pBuilderId, _In_ ULONG64 parentId = 0);
+
+    // ImportUDT():
+    //
+    // Imports the given UDT into the symbol builder.
+    //
+    HRESULT ImportUDT(_In_ ULONG symIndex, _Out_ ULONG64 *pBuilderId, _In_ ULONG64 parentId = 0);
+
+    // ImportEnum():
+    //
+    // Imports the given enum into the symbol builder.
+    //
+    HRESULT ImportEnum(_In_ ULONG symIndex, _Out_ ULONG64 *pBuilderId, _In_ ULONG64 parentId = 0);
+
+    // ImportBaseType():
+    //
+    // Imports the given base type into the symbol builder.
+    //
+    HRESULT ImportBaseType(_In_ ULONG symIndex, _Out_ ULONG64 *pBuilderId, _In_ ULONG64 parentId = 0);
+
+    // ImportTypeSymbol():
+    //
+    // Imports the given type symbol into the symbol builder.
+    //
+    HRESULT ImportTypeSymbol(_In_ ULONG symIndex, _Out_ ULONG64 *pBuilderId, _In_ ULONG64 parentId = 0);
+
+    // ImportBaseClass():
+    //
+    // Imports the given base class symbol into the symbol builder.
+    //
+    HRESULT ImportBaseClass(_In_ ULONG symIndex, _Out_ ULONG64 *pBuilderId, _In_ ULONG64 parentId);
+
+    //********************
+    // Data Symbol Import:
+    //
+
+    // ImportMemberData():
+    //
+    // Imports a data member (field) into the symbol builder.
+    //
+    HRESULT ImportMemberData(_In_ ULONG symIndex, _Out_ ULONG64 *pBuilderId, _In_ ULONG64 parentId);
+
+    // ImportConstantData():
+    //
+    // Imports constant data into the symbol builder.
+    //
+    HRESULT ImportConstantData(_In_ ULONG symIndex, _Out_ ULONG64 *pBuilderId, _In_ ULONG64 parentId);
+
+    // ImportDataSymbol():
+    //
+    // Imports the given data symbol into the symbol builder.
+    //
+    HRESULT ImportDataSymbol(_In_ ULONG symIndex, _Out_ ULONG64 *pBuilderId, _In_ ULONG64 parentId);
+
+    //********************
+    // Other Related:
+    //
+
+    // TagMatchesSearchCriteria():
+    //
+    // Check whether a symbol tag returned from DbgHelp matches the kind of symbol(s) we are looking for.
+    //
+    static bool TagMatchesSearchCriteria(_In_ ULONG tag, _In_ SvcSymbolKind searchKind);
+
+    //*************************************************
+    // DbgHelp Callback Handlers:
+    //
+
+    // LegacySymbolCallback():
+    //
+    // DbgHelp callbacks.
+    //
+    bool LegacySymbolCallback(_In_ HANDLE /*hProcess*/,
+                              _In_ ULONG actionCode,
+                              _In_ PVOID pCallbackData)
+    {
+        switch(actionCode)

```

Comment:
>actionCode [](http://example.com/codeflow?start=15&length=10)

You may want to handle CBA_XML_LOG if there is a way to forward this back to the client so that dbghelp logs get into the WinDbg log. Not sure if this layer has a way of funneling this up through DbgEng...

Reply:
We'd have to introduce new API to do that.  There's only ISvcDiagnosticLogging::Log(level, message, component, category) which will funnel to those logs...  but that's a flat text message...  we have no way of passing any of the XML data through...

Reply:
I can put a TODO comment here if we ever expand the diagnostic services available to a plug-in....

Reply:
That's fine.